be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -168,46 +177,98 @@ struct ProcessHashTableProbe {
     // the output block struct is same with mutable block. we can do more opt on it and simplify
     // the logic of probe
     // TODO: opt the visited here to reduce the size of hash table
+    template<bool is_left_semi_join, bool is_left_anti_join, bool is_inner_join>
     Status do_process(HashTableContext& hash_table_ctx, ConstNullMapPtr null_map,
                       MutableBlock& mutable_block, Block* output_block) {
         using KeyGetter = typename HashTableContext::State;
         using Mapped = typename HashTableContext::Mapped;
 
         KeyGetter key_getter(_probe_raw_ptrs, _join_node->_probe_key_sz, nullptr);
-
+    
         std::vector<uint32_t> items_counts(_probe_rows);
         auto& mcol = mutable_block.mutable_columns();
-
-        int right_col_idx = _join_node->_is_right_semi_anti ? 0 : _left_table_data_types.size();
-        int right_col_len = _right_table_data_types.size();
         int current_offset = 0;
 
         for (; _probe_index < _probe_rows;) {
-            // ignore null rows
             if constexpr (ignore_null) {
                 if ((*null_map)[_probe_index]) {
                     items_counts[_probe_index++] = 0;
                     continue;
                 }
             }
-
             int repeat_count = 0;
-            auto find_result =
-                    (*null_map)[_probe_index]
+            if constexpr (is_inner_join) {
+                if (!(*null_map)[_probe_index]) {
+                    auto find_result = key_getter.find_key(hash_table_ctx.hash_table, _probe_index, _arena);
```

Comment:
why not pretch here ?

Reply:
in my test ,wo found prefetch is time consuming，but not helpful for performance ,so maybe when mapped is more than 1, it is helpful.